packages/cli/src/commands/server/webSocketProxy.ts

```diff
@@ -54,20 +57,24 @@ function attachToServer(server, path) {
         return;
       }
       debuggerSocket = connection;
-      debuggerSocket.onerror = debuggerSocketCloseHandler;
-      debuggerSocket.onclose = debuggerSocketCloseHandler;
-      debuggerSocket.onmessage = ({data}) => send(clientSocket, data);
+      if (debuggerSocket) {
+        debuggerSocket.onerror = debuggerSocketCloseHandler;
+        debuggerSocket.onclose = debuggerSocketCloseHandler;
+        debuggerSocket.onmessage = ({data}) => send(clientSocket, data);
+      }
     } else if (url.indexOf('role=client') > -1) {
       if (clientSocket) {
-        clientSocket.onerror = null;
-        clientSocket.onclose = null;
-        clientSocket.onmessage = null;
+        clientSocket.onerror = () => {};
+        clientSocket.onclose = () => {};
+        clientSocket.onmessage = () => {};
         clientSocket.close(1011, 'Another client connected');
       }
       clientSocket = connection;
-      clientSocket.onerror = clientSocketCloseHandler;
-      clientSocket.onclose = clientSocketCloseHandler;
-      clientSocket.onmessage = ({data}) => send(debuggerSocket, data);
+      if (clientSocket) {
```

Comment:
we assume `connection` is defined on line 52, let's skip this check if possible

Reply:
changed the type of `connection` in this function argument and removed this check